### ISSUE TITLE
Write integration tests and simplify unit tests?

### ISSUE THREAD

**Issue by cvan** [mozilla/galaxy-api#341]
The unit tests test actual responses. (When you run [`npm test`](https://github.com/mozilla/galaxy-api/blob/c912cb47/package.json#L39), a new database is created (and destroyed first if needed), called [`galaxy-api_test`](https://github.com/mozilla/galaxy-api/blob/master/settings_test.js.dist), which is used strictly to run the tests.)

Anyway, it's swell that we're testing the views (controllers), but we should instead be asserting that the views are merely calling particular model calls (and have those model calls unit tested [issue #340]).

Fortunately, I've structured the `api.game` tests (and `lib` tests) such that `lab.experiment` calls are like separate "modules" (à la classes in python's `unittest`) and `lab.test` calls are separate "tests" (à la methods in python's `unittest`).

However, the state can unfortunately persist (i.e., leak) between tests, inside and outside the same "module." To avoid leaking between tests, I have added manually calls to clear all the rows in the `games` database table after each test. But we can go better. I've filed issue #337 to incorporate transactions in the tests.

Long story short, the unit tests should be testing things more in isolation.

It's tricky.

<hr>

Looking for feedback re: unit/integration/database testing strategies here. /cc @AdrianGaudebert @rfk @potch
